### PR TITLE
Add Exercise02

### PR DESCRIPTION
**Pull request by huynhngocminh2511**

### CODE REVIEW

File: Exercise02/Api/GitHub.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+        private WebClient request = new WebClient();
+
+        public List<User> GetUsers(string username)
+        {
+            request.Headers["User-Agent"] = "mytest";
+
+            var response = "";
+
+            try
```
Comment (voquanghoa) [inline Exercise02/Api/GitHub.cs]: Empty try-catch. Just remove it

File: Exercise02/Api/GitHub.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            });
+
+            return users.Users;
+        }
+
+        public Detail GetDetail(string url)
+        {
+            var response = "";
+
+            try
```
Comment (voquanghoa) [inline Exercise02/Api/GitHub.cs]: Remove this kind of try catch

File: Exercise02/Api/GitHub.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        public List<string> GetRepositories(string url)
+        {
+            var response = "";
+
+            try
+            {
+                request.Headers["User-Agent"] = "mytest";
+                response = request.DownloadString(url);
+            }
+            catch (WebException e)
```
Comment (voquanghoa) [inline Exercise02/Api/GitHub.cs]: Remove try catch

File: Exercise02/Details.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    public class Details : Activity
+    {
+        protected override void OnCreate(Bundle savedInstanceState)
+        {
+            base.OnCreate(savedInstanceState);
+
+            SetContentView(Resource.Layout.Details);
+
+            var detail = Detail.DetailCurrent;
+            FindViewById<TextView>(Resource.Id.tv_blog).Text = "Blog: " + detail.Blog;
```
Comment (voquanghoa) [inline Exercise02/Details.cs]: Extract to a method

File: Exercise02/Details.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            FindViewById<TextView>(Resource.Id.tv_company).Text = "Company: " + detail.Company;
+            FindViewById<TextView>(Resource.Id.tv_email).Text = "Email: " + detail.Email?.ToString() ?? "";
+            FindViewById<TextView>(Resource.Id.tv_location).Text = "Location: " + detail.Location;
+            FindViewById<TextView>(Resource.Id.tv_name).Text = "Name: " + detail.Name;
+
+            var listView = FindViewById<ListView>(Resource.Id.lv_repository);
+            var url = Intent.GetStringExtra("url");
+            var adapter = new ArrayAdapter<string>(this, Resource.Layout.Repository, GitHub.gitHub.GetRepositories(url));
```
Comment (voquanghoa) [inline Exercise02/Details.cs]: Extract method

File: Exercise02/MainActivity.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace Exercise02
+{
+    [Activity(Label = "Exercise02", Theme = "@android:style/Theme.Material.Light.DarkActionBar", MainLauncher = true)]
+    public class MainActivity : Activity
+    {
+        protected override void OnCreate(Bundle savedInstanceState)
+        {
+            base.OnCreate(savedInstanceState);
+
```
Comment (voquanghoa) [inline Exercise02/MainActivity.cs]: Extract method

File: Exercise02/Resources/values/Strings.xml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+  <string name="app_name">Exercise02</string>
+
+  <string name="tv_email">Email: </string>
```
Comment (voquanghoa) [inline Exercise02/Resources/values/Strings.xml]: Don't put prefix `tv_`.

File: Exercise02/Details.cs [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+using Android.Views;
+using Android.Widget;
+using Android.Support.V7.Widget;
+using Exercise02.Api;
+using Exercise02.Models;
+
+namespace Exercise02
+{
+    [Activity(Label = "Details", Theme = "@android:style/Theme.Material.Light.DarkActionBar")]
+    public class Details : Activity
```
Comment (voquanghoa) [inline Exercise02/Details.cs]: Activity name should be end withs `Activity`

File: Exercise02/Models/Detail.cs [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        public string Company { get; set; }
+
+        [JsonProperty("blog")]
+        public string Blog { get; set; }
+
+        [JsonProperty("location")]
+        public string Location { get; set; }
+
+        [JsonProperty("email")]
+        public object Email { get; set; }
```
Comment (voquanghoa) [inline Exercise02/Models/Detail.cs]: Object instead of string?

File: Exercise02/Details.cs [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        protected override void OnCreate(Bundle savedInstanceState)
+        {
+            base.OnCreate(savedInstanceState);
+
+            SetContentView(Resource.Layout.Details);
+
+            var detail = Detail.DetailCurrent;
```
Comment (voquanghoa) [inline Exercise02/Details.cs]: An activity should load the data for itself.

File: Exercise02/MainActivity.cs [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+            searchView.QueryTextSubmit += delegate
+            {
+                adapter.Users = GitHub.gitHub.GetUsers(searchView.Query);
+            };
+
+            adapter.ItemClick += (object sender, UserAdapterClickEventArgs e) =>
+            {
+                var intent = new Intent(this, typeof(Details));
+                intent.PutExtra("url", e.RepositoryUrl);
+                Detail.DetailCurrent = e.Detail;
```
Comment (voquanghoa) [inline Exercise02/MainActivity.cs]: Activity Details should reload the data

File: Exercise02/Models/Detail.cs [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+using Android.Runtime;
+using Android.Views;
+using Android.Widget;
+using Newtonsoft.Json;
+
+namespace Exercise02.Models
+{
+    public class Detail
+    {
+        public static Detail DetailCurrent = new Detail();
```
Comment (voquanghoa) [inline Exercise02/Models/Detail.cs]: Bad practic.
```C#
public static Detail DetailCurrent = new Detail();
```

File: Exercise02/Api/GitHub.cs (hunk 1 above)
Comment (huynhngocminh2511) [reply Exercise02/Api/GitHub.cs]: Done

File: Exercise02/Api/GitHub.cs (hunk 2 above)
Comment (huynhngocminh2511) [reply Exercise02/Api/GitHub.cs]: Done

File: Exercise02/Api/GitHub.cs (hunk 3 above)
Comment (huynhngocminh2511) [reply Exercise02/Api/GitHub.cs]: Done

File: Exercise02/Details.cs (hunk 8 above)
Comment (huynhngocminh2511) [reply Exercise02/Details.cs]: Done

File: Exercise02/Details.cs (hunk 10 above)
Comment (huynhngocminh2511) [reply Exercise02/Details.cs]: Done

File: Exercise02/Details.cs (hunk 4 above)
Comment (huynhngocminh2511) [reply Exercise02/Details.cs]: Done

File: Exercise02/Details.cs (hunk 5 above)
Comment (huynhngocminh2511) [reply Exercise02/Details.cs]: Done

File: Exercise02/MainActivity.cs (hunk 11 above)
Comment (huynhngocminh2511) [reply Exercise02/MainActivity.cs]: Done

File: Exercise02/Models/Detail.cs (hunk 12 above)
Comment (huynhngocminh2511) [reply Exercise02/Models/Detail.cs]: Done

File: Exercise02/Resources/values/Strings.xml (hunk 7 above)
Comment (huynhngocminh2511) [reply Exercise02/Resources/values/Strings.xml]: Done

File: Exercise02/MainActivity.cs (hunk 6 above)
Comment (huynhngocminh2511) [reply Exercise02/MainActivity.cs]: Done

File: Exercise02/Models/Detail.cs (hunk 9 above)
Comment (huynhngocminh2511) [reply Exercise02/Models/Detail.cs]: Done

File: Exercise02/Adapters/UserAdapter.cs [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+using Android.Views;
+using Android.Widget;
+using Android.Support.V7.Widget;
+using System.Collections.Generic;
+using Exercise02.Models;
+using Square.Picasso;
+
+namespace Exercise02.Adapters
+{
+    class UserAdapter : RecyclerView.Adapter
```
Comment (voquanghoa) [inline Exercise02/Adapters/UserAdapter.cs]: Split file

File: Exercise02/Api/GitHub.cs [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+        public bool CheckForInternetConnection(Context context)
+        {
+            try
+            {
+                SetHeader();
+                request.DownloadString(root);
+            }
+            catch(WebException e)
+            {
+                if (e.Status == WebExceptionStatus.NameResolutionFailure || e.Status == WebExceptionStatus.ReceiveFailure)
```
Comment (voquanghoa) [inline Exercise02/Api/GitHub.cs]: Wrong way to handle exception

File: Exercise02/Models/User.cs [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+using Android.Content;
+using Android.OS;
+using Android.Runtime;
+using Android.Views;
+using Android.Widget;
+using Newtonsoft.Json;
+
+namespace Exercise02.Models
+{
+    public class ListOfUser
```
Comment (voquanghoa) [inline Exercise02/Models/User.cs]: Split file

File: Exercise02/Adapters/UserAdapter.cs (hunk 13 above)
Comment (huynhngocminh2511) [reply Exercise02/Adapters/UserAdapter.cs]: Done

File: Exercise02/Api/GitHub.cs (hunk 14 above)
Comment (huynhngocminh2511) [reply Exercise02/Api/GitHub.cs]: Done

File: Exercise02/Models/User.cs (hunk 15 above)
Comment (huynhngocminh2511) [reply Exercise02/Models/User.cs]: Done

**Pull request merged**